Makefile

```diff
@@ -821,6 +813,56 @@ ifneq ($(word 2,$(MAKECMDGOALS)),)
 export ENABLE_MSG_EXTRA := yes
 endif
 
+uavobjects_armsoftfp: TARGET=uavobjects_armsoftfp
+uavobjects_armsoftfp: OUTDIR=$(BUILD_DIR)/$(TARGET)
+uavobjects_armsoftfp: uavobjects
+	$(V1) mkdir -p $(OUTDIR)/dep
+	$(V1) cd $(ROOT_DIR)/flight/uavobjectlib && \
+		$(MAKE) -r --no-print-directory \
+		TCHAIN_PREFIX="$(ARM_SDK_PREFIX)" \
+		REMOVE_CMD="$(RM)" \
+		\
+		MAKE_INC_DIR=$(MAKE_INC_DIR) \
+		ROOT_DIR=$(ROOT_DIR) \
+		TARGET=$(TARGET) \
+		OUTDIR=$(OUTDIR) \
+		\
+		PIOS=$(PIOS) \
```

Comment:
e.g. picking the first one of these I find in the makefile

```
ut_$(1)_%: $$(UT_OUT_DIR)
        $(V1) mkdir -p $(UT_OUT_DIR)/$(1)
        $(V1) cd $$(UT_ROOT_DIR) && \
                $$(MAKE) -r --no-print-directory \
                BUILD_TYPE=ut \
                BOARD_SHORT_NAME=$(1) \
                TCHAIN_PREFIX="" \
                REMOVE_CMD="$(RM)" \
                \
                MAKE_INC_DIR=$(MAKE_INC_DIR) \
                ROOT_DIR=$(ROOT_DIR) \
                BOARD_ROOT_DIR=$$(BOARD_ROOT_DIR) \
                BOARD_INFO_DIR=$$(BOARD_ROOT_DIR)/board-info \
                TARGET=$$(TARGET) \
                OUTDIR=$$(OUTDIR) \
                \
                PIOS=$(PIOS) \
                OPUAVOBJ=$(OPUAVOBJ) \
                OPUAVTALK=$(OPUAVTALK) \
                OPMODULEDIR=$(OPMODULEDIR) \
                FLIGHTLIB=$(FLIGHTLIB) \
                SHAREDAPIDIR=$(SHAREDAPIDIR) \
                \
                GTEST_DIR=$(GTEST_DIR) \
                \
                $$*
```
A lot of these are dead.  BUILD_TYPE/BOARD_SHORT_NAME are abused.  PIOS is dead / wrong for unit test targets.  Some of the other libs are dead.

It takes a lot of work to make sure these are really dead and I question the benefits from narrowing the scoping of a lot of these things that I think it's fine for all flight makefiles to know.

For now, I'll cut down this new one when I get to it, but I hope we could reach agreement to just export some of the more frequently used ones of these and cut down the boilerplate.